src/tracked_unique.rs

```diff
@@ -0,0 +1,152 @@
+//! A small change tracker to wrap around a unique so you can determine if the value changes between updates.
+use tracing::trace_span;
+
+use crate::prelude::*;
+
+use core::any::type_name;
+use std::{fmt, ops::Deref, ops::DerefMut};
+
+pub struct TrackedValue<T: 'static>(InnerTrackedState, T);
+
+pub struct TrackedMut<'a, T: 'static>(UniqueViewMut<'a, TrackedValue<T>>);
+pub struct Tracked<'a, T: 'static>(UniqueView<'a, TrackedValue<T>>);
+
+impl<'a, T: 'static + Send + Sync> Borrow<'a> for Tracked<'a, T> {
+    fn borrow(
+        all_storages: &'a AllStorages,
+        all_borrow: Option<SharedBorrow<'a>>,
+    ) -> Result<Self, error::GetStorage>
+    where
+        Self: Sized,
+    {
+        Ok(Tracked(Borrow::borrow(all_storages, all_borrow)?))
+    }
+}
+
+unsafe impl<'a, T: 'static + Send + Sync> BorrowInfo for Tracked<'a, T> {
```

Comment:
This is an interesting concept; but I don't understand what `BorrowInfo` does, yet. Note to self: Need to look into this at some point.

Reply:
Don't look too closely right now since it's being reworked a bit by @leudz.

Reply:
`BorrowInfo` collects information about what each system borrows, this is used by `WorkloadBuilder` to decide scheduling inside a workload.
The `borrow_info` function was previously in `Borrow`, I moved it to its own trait to make it `unsafe`. I will add a way to bypass locking and trust the scheduler.
But the scheduler can only do correctly its job if `BorrowInfo` is implemented correctly.